### ISSUE TITLE
Allow use of timers outside of `@timeit` macro

### ISSUE THREAD

**Issue by dm3** [KristofferC/TimerOutputs.jl#37]
Hey, great library you've built! However, I've stumbled into an issue almost immediately. It's impossible to properly time blocks which do an early return or continue. There's an issue - #34 - that highlights the same problem. 

I understand that the solution involving parsing of the timed expression to identify early exits is complicated, therefore I'd like to suggest another one - reifying the "timing block" instances. For example, in order to time an early-return block you could do:

```
tos = TimerOutputs()

timer = TimerOutputs.start!(tos, "block")
try
  if compute_something() > 5
    TimerOutputs.stop!(timer)
    return false
  end
  ...
finally
  TimerOutputs.stop!(timer)
end
```

On one hand, this transfers the timer lifecycle management responsibility to the user, which is a worse experience than a magic `@timeit` macro. On the other hand, this approach makes timing early returning and (potentially) asynchronous code possible without having to complicate the `@timeit` macro. The macro can continue to work the same way it worked before.

What do you think?

**Comment by KristofferC** [comment]
That's exactly what to `do` syntax does currently e.g in the README:

```
# a function version with `do` syntax for exception safety
function i_will_throw()
    timeit(to, "throwing") do
        sleep(0.5)
        throw(error("this is fine..."))
        print("nope")
    end
end
```

If you look at:

https://github.com/KristofferC/TimerOutputs.jl/blob/6411148543f92e5430fefac943f09c91add0cb53/src/TimerOutput.jl#L198-L212

you can see it is implemented in a `try`, `finally` fashion.

**Comment by dm3** [comment]
You're right! I haven't noticed that function! 

What do you think about the possibility to time async code? After looking at the internals this would entail changing the timer stack mechanism. Do you think it's out of scope for TimerOutputs?

**Comment by KristofferC** [comment]
If it can be done without sacrificing too much performance in the single task case, it would be a nice thing to have.

**Comment by KristofferC** [comment]
This is fixed on 0.7 now, without having to use the `do` syntax. Async stuff is another issue :)
